### ISSUE TITLE
"Directories are not supported" with plugin version 1.2.0

### ISSUE THREAD

**Issue by NicolaSpreafico** [GoogleCloudPlatform/app-maven-plugin#144]
My plugin is configured like this:
```xml
<plugin>
    <groupId>com.google.cloud.tools</groupId>
    <artifactId>appengine-maven-plugin</artifactId>
    <version>1.0.0</version>
    <configuration>
        <deploy.version>${app.version}</deploy.version>
        <deploy.project>${app.id}</deploy.project>
        <deploy.promote>False</deploy.promote>
        <deployables>
            <param>target/appengine-staging/app.yaml</param>
            <param>target/appengine-stagine/cron.yaml</param>
        </deployables>
    </configuration>
</plugin>
```

Using the `com.google.cloud.tools:appengine-maven-plugin:deploy` goal the deploy works fine, with the following command:
```
INFORMAZIONI: submitting command: C:\Users\Nicola\AppData\Local\Google\Cloud SDK\google-cloud-sdk\bin\gcloud.cmd app deploy C:\Users\Nicola\Noovle\git\ape-core\project\target\appengine-staging\app.yaml C:\Users\Nicola\Noovle\git\ape-core\project\target\appengine-staging\cron.yaml --no-promote --version test-deploy --project ape-base --quiet
```

Upgrading the plugin version from `1.0.0` to `1.2.0` (and w/o touching anything else), during the deploy I get this error instead
```
INFORMAZIONI: submitting command: C:\Users\Nicola\AppData\Local\Google\Cloud SDK\google-cloud-sdk\bin\gcloud.cmd app deploy C:\Users\Nicola\Noovle\git\ape-core\project\target\appengine-staging\app.yaml C:\Users\Nicola\Noovle\git\ape-core\project\target\appengine-staging\cron.yaml C:\Users\Nicola\Noovle\git\ape-core\project\target\appengine-staging --no-promote --version test-deploy --project ape-base

[INFO] GCLOUD: ERROR: Directories are not supported [C:\Users\Nicola\Noovle\git\ape-core\project\target\appengine-staging].  You must provide explicit yaml files.

[INFO] GCLOUD: ERROR: (gcloud.app.deploy) Errors occurred while parsing the App Engine app configuration.
```

Please note the generated command, after the `app.yaml` and `cron.yaml` path, there is an additional 
`C:\Users\Nicola\Noovle\git\ape-core\project\target\appengine-staging` folder

**Comment by patflynn** [comment]
@loosebazooka this looks like a high priority issue.

**Comment by joaoandremartins** [comment]
I'm wondering if this has to do with https://github.com/GoogleCloudPlatform/appengine-plugins-core/pull/318

**Comment by meltsufin** [comment]
The bug is here `DeployMojo.execute`:
```java
    File appYamlFile = new File(stagingDirectory + "/app.yaml");
    if (deployables.size() == 0 && appYamlFile.exists()) {
      deployables.add(appYamlFile);
    } else {
      deployables.add(stagingDirectory);
    }
```
I'll fix it.

**Comment by patflynn** [comment]
Can we add a test for this?

**Comment by meltsufin** [comment]
I did. See PR.

**Comment by patflynn** [comment]
great. Thanks!

**Comment by meltsufin** [comment]
@NicolaSpreafico The fixed version 1.2.1 is in Maven Central now. Thanks for reporting the bug!
